app/src/main/java/de/triology/cas/services/RegistryEtcd.java

```diff
@@ -48,6 +50,57 @@ public RegistryEtcd(EtcdClient etcd) {
         this.etcd = etcd;
     }
 
+
+    /**
+     * Retrieves all CAS Services Accounts which are currently registered in etcd.
+     *
+     * @return a list containing the identifier for all registered service accounts of cas
+     */
+    @Override
+    public List<CesServiceData> getInstalledOAuthCASServiceAccounts(ICesServiceFactory factory) {
+        log.debug("Get CAS-OAuth service accounts from registry");
+        try {
+            List<EtcdKeysResponse.EtcdNode> nodes = etcd.getDir(CAS_SERVICE_ACCOUNT_DIR).send().get().getNode().getNodes();
+            return extractOAuthClientsFromSADir(nodes, factory);
+        }
+        catch (EtcdException e) {
+            if (e.isErrorCode(EtcdErrorCode.KeyNotFound)) {
+                return new ArrayList<>();
+            } else {
+                log.warn("Failed to getInstalledOAuthCASServiceAccounts: ", e);
+                throw new RegistryException(e);
+            }
+        }
+        catch (IOException | EtcdAuthenticationException | TimeoutException e) {
+            log.error("Failed to getInstalledOAuthCASServiceAccounts: ", e);
+            throw new RegistryException(e);
+        }
+    }
+
+    /**
+     * Iterates over all available etcd-Keys of CASs service accounts.
+     *
+     * @param nodesFromEtcd a list containing all child nodes of the `service_accounts` directory of the cas in the etcd
+     * @return a list containing the identifier for all registered service accounts of cas
+     */
+    private List<CesServiceData> extractOAuthClientsFromSADir(List<EtcdKeysResponse.EtcdNode> nodesFromEtcd, ICesServiceFactory factory) {
+        log.debug("Entered extractOAuthClientsFromSADir");
+        List<CesServiceData> serviceAccountNames = new ArrayList<>();
```

Comment:
Why the variable is named `serviceAccountNames`? This implies that it is a list of names, but in fact it contains full service data.

Reply:
I changed the name to serviceDataList